ophyd/areadetector/base.py

```diff
@@ -100,13 +100,11 @@ def forward(self, value):
 
     def inverse(self, value):
         """Shape the flat array to send as a result of ``.get``"""
-        print(self.derived_shape, self._shape, self.derived_ndims)
         array_shape = self.derived_shape[:self.derived_ndims]
         if not any(array_shape):
             raise RuntimeError(f"Invalid array size {self.derived_shape}")
 
-        array = self._derived_from.get()
-        return np.array(array).reshape(array_shape)
+        return np.array(value[:np.prod(array_shape)]).reshape(array_shape)
```

Comment:
Shouldn't it be `np.array(array[...` using the `array` from L108? Not sure there the input `value` was used before.

Reply:
What type is `value` here? If it already an array (at least sometimes, in some circumstances) using `np.asarray` in place of `np.array` will save us from making a copy.

Reply:
> Shouldn't it be `np.array(array[...` using the `array` from L108? Not sure there the input `value` was used before.

The array value is actually passed into this method as the `value` arg; there's no need to get it again, so I've removed that line.